### ISSUE TITLE
XLA gradient of `reduce_max`/`reduce_min` silently returns 0 instead of NaN when input contains NaN

### ISSUE THREAD

**Issue by wuyii8941** [tensorflow/tensorflow#118673]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

TensorFlow 2.22.0-dev20260429

### Custom code

Yes

## Summary

Under XLA (`jit_compile=True`), the gradient of `tf.reduce_max` and `tf.reduce_min` returns all-zeros when the input contains NaN, while eager mode correctly returns NaN gradients. The forward pass agrees (both return NaN), but the backward pass silently drops the NaN signal.

This is a gradient-only bug: NaN in input → correct NaN forward result → **zero gradient** under XLA, instead of the expected NaN gradient.

## Scope

Affects **all array sizes** (not size-dependent):

| Input | Forward (both agree) | Eager gradient | XLA gradient |
|---|---|---|---|
| `[NaN, NaN, NaN]` | NaN | `[NaN, NaN, NaN]` | **`[0, 0, 0]`** |
| `[1, 2, NaN]` | NaN | `[NaN, NaN, NaN]` | **`[0, 0, 0]`** |
| `[NaN, 1, 2]` | NaN | `[NaN, NaN, NaN]` | **`[0, 0, 0]`** |
| 5 elements with 1 NaN | NaN | all NaN | **all 0** |
| 8 elements with 1 NaN | NaN | all NaN | **all 0** |

Also affects `tf.reduce_min`:

```python
x = tf.constant([float('nan')] * 3)
# reduce_min gradient: eager = [nan, nan, nan], XLA = [0, 0, 0]
```

## Practical Impact

In ML training with XLA compilation, NaN in a batch should cause NaN gradients, signaling that something is wrong upstream. With this bug, XLA silently produces zero gradients instead, making the training step a no-op for that loss component. This hides numerical issues rather than propagating them:

```python
x = tf.constant([1.0, float('nan'), 3.0, 2.0])

@tf.function(jit_compile=True)
def max_loss_grad(x):
    with tf.GradientTape() as t:
        t.watch(x)
        loss = tf.reduce_max(x)  # NaN
    return t.gradient(loss, x)

print(max_loss_grad(x).numpy())
# [0. 0. 0. 0.] — silently zero, hiding the NaN problem
```

Any training loop using `reduce_max` or `reduce_min` in the loss (e.g., max-margin losses, Chebyshev distance) will silently produce wrong gradients when NaN appears in the data.

## Root Cause

XLA computes the gradient of `reduce_max` using an indicator function: `gradient = (input == max_value) * upstream_gradient`. When the max value is NaN, the comparison `input == NaN` evaluates to `False` for all elements (per IEEE 754), so the gradient becomes 0 everywhere.

The eager implementation handles this correctly by propagating NaN through the gradient computation.

## Cross-framework check

JAX (0.9.2) has the **same bug**: `jax.grad(jnp.max)` returns `[0, 0, 0]` under `jax.jit` when input contains NaN, while JAX eager returns `[NaN, NaN, NaN]`. This confirms the root cause is in XLA core's gradient implementation, not in TensorFlow's XLA bridge.

```python
import jax, jax.numpy as jnp
grad_max = lambda x: jax.grad(lambda x: jnp.max(x).sum())(x)
x = jnp.array([1.0, 2.0, float('nan')])
print(grad_max(x))            # [nan nan nan]
print(jax.jit(grad_max)(x))   # [0. 0. 0.]  — same bug
```

## Environment

- TensorFlow 2.22.0-dev20260429
- JAX 0.9.2: **same bug** (eager NaN vs jit 0)
- Affects both CPU and GPU XLA
- Forward pass: agrees (both return NaN)
- Gradient: **diverges** (eager=NaN, XLA=0)

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

x = tf.constant([1.0, 2.0, float('nan')])

# Eager: gradient = [nan, nan, nan] (correct)
with tf.GradientTape() as t:
    t.watch(x)
    y = tf.reduce_max(x)
print(t.gradient(y, x).numpy())  # [nan nan nan]

# XLA: gradient = [0, 0, 0] (WRONG)
@tf.function(jit_compile=True)
def grad_max(x):
    with tf.GradientTape() as t:
        t.watch(x)
        y = tf.reduce_max(x)
    return t.gradient(y, x)
print(grad_max(x).numpy())  # [0. 0. 0.]


Observed on TensorFlow 2.22.0-dev20260429.
```

### Relevant log output

```shell

```

**Comment by dhantule** [comment]
HI @wuyii8941, 

Thanks for reporting this and sharing the details.
We’ve received the issue and are taking a look. We’ll follow up here once we have more information or if we need any additional inputs.
Appreciate your patience.
